core/math/a_star.cpp

```diff
@@ -217,7 +217,7 @@ int AStar::get_closest_point(const Vector3 &p_point) const {
 	for (const Map<int, Point *>::Element *E = points.front(); E; E = E->next()) {
 
 		real_t d = p_point.distance_squared_to(E->get()->pos);
-		if (closest_id < 0 || d < closest_dist) {
+		if ((closest_id < 0 || d < closest_dist) && E->get()->enabled == true) {
```

Comment:
No need to use `== true`, just `E->get()->enabled` would already be `true` or `false` by itself.

It can be improved further by returning early without doing distance calculations in this case though, with:
```cpp
if (!E->get()->enabled) {
	continue;
}

if (closest_id < 0 || d < closest_dist) {
...
}
```

Reply:
With boolean values while if (bool) does work, is there a reason to use that instead of if(bool==value)? Having the value might be redundant but it does improve readability.

Reply:
That's probably subjective, but I don't think `== true` or `== false` checks improve readability.

And they are indeed redundant and some compilers or static analyzers will raise warnings when they are used. See e.g.:
```
bool enabled = true;
if (enabled == true) {}
```
The condition will be parsed as `if (true == true)`, and then reduced to `if (true)`, since (`true == true` happens to be a `true` statement). But we already had the `true` statement from the start in `enabled` itself, so we don't need an extra `bool == bool` operation.

Reply:
Ok, I'll make that change.

Reply:
See https://softwareengineering.stackexchange.com/questions/12807/make-a-big-deal-out-of-true (I don't agree with all the snarky comments there, but it raises some relevant points on why `some_bool == true` is not a good practice).

Reply:
Thanks for the link, I went ahead and changed that, and moved the check up in the loop to avoid the distance check.